### PR TITLE
Feature/engagement workflow2 Issue #255

### PR DESCRIPTION
**Pull request by elisabeth-sorrell**
- add in project step requirements for each engagement status
- fix the engagement workflow tests to have the required project step for the engagement status advancements

### CODE REVIEW

**Comment by elisabeth-sorrell** [comment]
@CarsonF ready for your review

**Comment by elisabeth-sorrell** [comment]
@CarsonF can you review when you get a chance?

File: test/engagement-workflow.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    const internProject = await createProject(app, {
+      type: ProjectType.Internship,
+    });
+    const internEngagement = await createInternshipEngagement(app, {
+      projectId: internProject.id,
+    });
+    expect(internEngagement.status.value).toBe(EngagementStatus.InDevelopment);
+  });
+
+  it('Translation: should test engagement status Active when Project is made Active', async function () {
```
Comment (CarsonF) [inline test/engagement-workflow.e2e-spec.ts:122]: FYI you can nest describe() calls to help with structuring.

```tsx
describe('should test engagement status Active when Project is made Active', () => {
  it('translation', () => { ... });
  it('internship', () => { ... });
});

**Comment by elisabeth-sorrell** [comment]
@michaelmarshall or @CarsonF resolved comments and tests passed. Ready to merge at your convenience.

**Pull request merged**